Готовая первая работа Review

KsuyVas opened the pull request:
Отправляю на Review первую проектную работу "С чистого листа"

KsuyVas committed:
Первый проект. Первая попытка

KsuyVas committed:
Первый проект

KsuyVas committed:
Delete style.css

KsuyVas committed:
Первый проект С чистого листа

KsuyVas committed:
Add files via upload

KsuyVas committed:
Первый проект С чистого листа

KsuyVas committed:
Проверка

KsuyVas committed:
Выполнила первый проект С чистого листа

Avaakum commented on styles/style.css:
```diff
@@ -0,0 +1,321 @@
+.content-section_list {
+    display: flex;
+    padding-left: 20px;
+    padding-right: 20px;
```
Элемент шире чем макет https://disk.yandex.ru/i/hOkuUHk_azjgLw Лишний отступ, он создаёт горизонтальный скрол

Avaakum commented on index.html:
```diff
@@ -0,0 +1,229 @@
+
+<head>
+    <meta charset="UTF-8">
+    <title>С чистого листа</title>
```
Не задана meta[name=viewport], сайт будет некорректно отображаться при различных ширинах и на различных устройствах. Она скорее всего была или в стартовом наборе или в рекомендациях

https://disk.yandex.ru/i/xr-tGVZAIwPK8Q


 - https://developer.mozilla.org/ru/docs/Glossary/Viewport

Avaakum commented on styles/style.css:
```diff
@@ -0,0 +1,321 @@
+    margin: 0;
+    font-family: "Inter", sans-serif;
+    font-weight: 400;
+    background-color: white;
```
Цвет фона должен быть  #1e1e1e

Avaakum commented on index.html:
```diff
@@ -0,0 +1,229 @@
+    <meta charset="UTF-8">
+    <title>С чистого листа</title>
+    <link rel="stylesheet" href="./fonts/fonts.css">
+    <link rel="stylesheet" href="./styles/style.css">
```
---

**Отлично:**

Правильный порядок подключения стилей. Шрифты должны быть подключены первыми, потом общие стили(сброс), потом кастомные. При таком порядке подключении шрифты могут быть использованы в двух последующих файлах.  А так же сброс стилей должен быть до подключения своих стилей из-за специфики CSS, чтобы `global.css` не переопределил никаких наших стилей.

Avaakum commented on index.html:
```diff
@@ -0,0 +1,229 @@
+
+    <header class="content-section content-section_theme_dark header">
+
+        <a href="/"> <img src="./images/logo.svg" alt="Логотип Вёрстки" class="header_logo_image"></a>
```
Логотип должен быть ссылкой без атрибута href(что бы по ссылке не было перехода) в соответствии с уроком "рекомендации к проекту" для улучшения семантики страницы.

Avaakum commented on index.html:
```diff
@@ -0,0 +1,229 @@
+                    <span>страхом чистого листа</span>
+                </h2>
+
+                <time class="intro_date" datetime="2022-06-05">05.06.2022 г.</time>
```
---

**Отлично:**

Отличная семантика

Avaakum commented on index.html:
```diff
@@ -0,0 +1,229 @@
+        </section>
+
+        <section class="content-section content-section_theme_bright reasons">
+            <div>
```
Лишний div

Avaakum commented on index.html:
```diff
@@ -0,0 +1,229 @@
+
+        <section class="content-section content-section_theme_bright reasons">
+            <div>
+                <h3 class="content-section_subtitle">
```
- Уровень заголовка не соответствует структуре страницы. Это заголовок секции, по структуре будет h2


---


- Про заголовки и подзаголовки на примере другой страницы с секциями, хэдером, и футером: у нас есть **один заголовок h1 на странице**, в каждой секции есть свой заголовок h2, если там есть другие элементы с заголовками ниже по иерархии, то они будут h3. Если например в секции был бы заголовок h2 и подзаголовок h3, то элементы с заголовками ниже по иерархии в этой же секции были бы уже h4.

KsuyVas replied:
Изначально я хотела расставить заголовки, как вы писали, но потом увидела в рекомендациях: "_Для заголовка используйте тег, описывающий корректную структуру: это всё одна статья, заголовок в секции reasons должен быть ниже уровнем, чем в секции intro."_ и это сбило меня с толку)


там в секции INTRO заголовок "8 способов борьбы со страхом чистого листа". который вообще не бросается в глаза из-за своих маленьких размеров.

Avaakum commented on index.html:
```diff
@@ -0,0 +1,229 @@
+                потока.
+                Причин, почему первые усилия даются так сложно, может быть очень много. Вот основные:
+            </p>
+            <ul class="content-section_list">
```
---

**Отлично:**

Отлично, использован список для повторяющихся элементов

Avaakum commented on index.html:
```diff
@@ -0,0 +1,229 @@
+        </section>
+
+        <section class="content-section content-section_theme_dark how-to">
+            <div>
```
Лишний div

Avaakum commented on index.html:
```diff
@@ -0,0 +1,229 @@
+                        </div>
+
+                        <div class="card__content">
+                            <h4 class="card__title">Начать любыми словами</h4>
```
- Уровень заголовка не соответствует структуре страницы. По структуре тут и далее будет h3

Avaakum commented on index.html:
```diff
@@ -0,0 +1,229 @@
+            © VЁRSTKA
+        </div>
+        <ul class="footer_links">
+            <li>ВК</li>
```
Внутри каждого элемента списк должна быть ссылка с атрибутом href

Avaakum commented on index.html:
```diff
@@ -0,0 +1,229 @@
+
+    <footer class="content-section content-section_theme_dark footer">
+
+        <div class="footer_copy">
```
Используйте для текста тег параграф

Avaakum reviewed (changes requested):
Развернуть общий комментарий ⬇︎

Доброго времени суток. У вас получилась хорошая работа!

_Что понравилось:_

- Хорошая структура HTML
- похоже на макет
- хорошие стили
- используется семантика
- Используются списки


_Что необходимо исправить:_

- **Все комментарии в коде, кроме комментариев «Можно лучше» и «Отлично»**

- https://disk.yandex.ru/i/ATWDbYqxsanIdg Нужно указать альтернативный шрифт (тип засечек для выбора шрифта браузера без засечек по-умолчанию значение sans-serif и для шрифта с засечками тип serif) для каждого значения font-family в проекте, кроме файла подключения шрифтов, где это св-во задаёт только название подключаемого шрифта
    - https://doka.guide/css/font-family


- Необходимо подправить форматирование HTML(используется 4 пробела вместо 2х, код очень разрастается). Для того чтобы отформатировать `index.html` можно воспользоваться авто-форматтером, например  [HTML Formatter](https://www.freeformatter.com/html-formatter.html). Для того, чтобы отступы были в соответствии со стандартами написания кода, то нужно выбрать пункт "отступ в 2 пробела" (2 spaces). Или для того, чтобы улучшить структуру отступов и переносов элементов к `index.html` можно воспользоваться встроенной функциейVSCode. В любом месте нажать правой кнопкой мыши и выбрать "Форматировать документ"
   - [VS-code HTML formatting settings](https://code.visualstudio.com/docs/languages/html#_formatting)


_Что можно улучшить:_

- Все комментарии "Можно лучше"

- Селекторы в файле стилей должны располагаться в порядке их появления в html

- В любом месте в файлах CSS в проекте(кроме самого CSS файла подключения шрифтов, где указывается только название подключаемого шрифта), где указано свойство  `font-family`  помимо основного шрифта так же должен быть указан альтернативный шрифт и тип засечек. Это делается для того, что если по каким-то причинами основной шрифт не загрузится, то внешний вид страницы не будет изменен очень сильно.  Порядок написания шрифтов: Основной шрифт, [Веб-безопасный альтернативный шрифт(по ссылке список этих шрифтов)](https://www.cssfontstack.com/), Тип засечек. Пример ` font-family: Inter, Arial, sans-serif; ` 

- Желательно использовать все классы по БЭМ. 
  - https://ru.bem.info/methodology/quick-start/
  - Проверить ошибки БЭМ можно на  https://nglazov.github.io/bem-validator-page/
  - струтуру можно проверить на https://yoksel.github.io/html-tree/

- Можно использовать атрибуты `target="_blank"` для ссылок на сторонние ресурсы для того чтобы пользователю не приходилось заново открывать сайт проекта
- Для более точной корректировки размеров текстов и расстояний между элементами можно воспользоваться [расширением PixelPerfect](https://chrome.google.com/webstore/detail/perfectpixel-by-welldonec/dkaagdgjmgdmbnecmcefdhjekcoceebi) для Chrome 


**Напоминание:** После выполнения всех критических замечаний работа будет принята. Предложения по улучшению выполняются по желанию.

Желаю успеха!

KsuyVas commented:
Доброго!) Большое спасибо за комментарии! Буду изучать и исправлять )

KsuyVas committed:
проект с Чистого листа . Второе ревью

Внесла все критические поправки согласно комментариям. И дополнительные, кроме описания классов согласно БЭМ. Отправляю работу на повторное ревью.

Avaakum reviewed (approved):
Развернуть общий комментарий ⬇︎

Здравствуйте! Рад вам сообщить, что ваша работа принята. Сейчас страничка выглядит прекрасно, всё работает отлично. Отличная семантика и структура. Желаю вам удачи в следующих спринтах и проектах! 

---

_Что необходимо исправить:_

- https://disk.yandex.ru/i/FrooBg5MPgLAQQ  Когда в названии шрифта есть пробелы, то оно должно быт в кавычках или браузер не понимает это значение


---




Ссылки:

- Можно использовать неразрывные пробелы для того чтобы текст не переносился в неожиданных местах. Неразрывные пробелы заменяют обычные, если нужно чтобы какая-то фраза не переносилась, например не нужно чтобы "на собаках" раскидывалось по разным строчкам при некоторых ширинах экранов, тогда `«на&nbsp;собаках»`. Но если страница с адаптацией, то всегда стоит проверять как текст с пробелами выглядит на всех ширинах.
- Про альтернативный текст для картинок https://habr.com/ru/post/341810/
- [Ссылка про подгрузку шрифтов](https://only-to-top.ru/blog/tools/2019-03-13-nastrojte-pokaz-vsego-teksta-vo-vremya-zagruzki-veb-shriftov.html) 
- Для более точной корректировки размеров текстов и расстояний между элементами можно воспользоваться [расширением PixelPerfect](https://chrome.google.com/webstore/detail/perfectpixel-by-welldonec/dkaagdgjmgdmbnecmcefdhjekcoceebi) для Chrome 



Несколько  ссылок на материал по теме нативных CSS переменных от авторов практикума:

- [1 часть](https://tproger.ru/articles/custom-properties-in-css-part-1/)

- [2 часть](https://tproger.ru/articles/custom-properties-in-css-part-2/)

- [3 часть](https://tproger.ru/articles/custom-properties-in-css-part-3/)

KsuyVas commented:
Здравствуйте,  большое спасибо!)

KsuyVas committed:
Add files via upload
